Modules/_xxsubinterpretersmodule.c

```diff
@@ -1972,18 +2204,41 @@ _run_script_in_interpreter(PyInterpreterState *interp, const char *codestr,
    the data that we need to share between interpreters, so it cannot
    hold PyObject values. */
 static struct globals {
+    int module_count;
     _channels channels;
-} _globals = {{0}};
+} _globals = {0};
 
 static int
-_init_globals(void)
+_globals_init(void)
 {
-    if (_channels_init(&_globals.channels) != 0) {
-        return -1;
+    // XXX This isn't thread-safe.
+    _globals.module_count++;
+    if (_globals.module_count > 1) {
+        // Already initialized.
+        return 0;
     }
+
+    assert(_globals.channels.mutex == NULL);
+    PyThread_type_lock mutex = PyThread_allocate_lock();
+    if (mutex == NULL) {
+        return ERR_CHANNELS_MUTEX_INIT;
+    }
+    _channels_init(&_globals.channels, mutex);
     return 0;
 }
 
+static void
+_globals_fini(void)
+{
+    // XXX This isn't thread-safe.
+    if (_globals.module_count > 1) {
+        return;
+    }
+    _globals.module_count--;
```

Comment:
`module_count` will never reach 0. You'll need to do something like this:

```suggestion
    if (_globals.module_count-- > 1) {
        return;
    }
```

Reply:
good catch!

Reply:
fixed